map/routing_manager.cpp

```diff
@@ -345,56 +357,43 @@ RouterType RoutingManager::GetLastUsedRouter() const
   }
 }
 
-void RoutingManager::SetRouterImpl(routing::RouterType type)
+void RoutingManager::SetRouterImpl(RouterType type)
 {
   auto const indexGetterFn = m_callbacks.m_featureIndexGetter;
-  ASSERT(indexGetterFn, ());
-  std::unique_ptr<IRouter> router;
-  std::unique_ptr<OnlineAbsentCountriesFetcher> fetcher;
+  CHECK(indexGetterFn, ("Type:", type));
+
+  VehicleType const vehicleType = GetVehicleType(type);
 
   auto const countryFileGetter = [this](m2::PointD const & p) -> std::string {
     // TODO (@gorshenin): fix CountryInfoGetter to return CountryFile
     // instances instead of plain strings.
     return m_callbacks.m_countryInfoGetter().GetRegionCountryId(p);
   };
 
-  auto numMwmIds = make_shared<routing::NumMwmIds>();
+  auto numMwmIds = make_shared<NumMwmIds>();
   m_delegate.RegisterCountryFilesOnRoute(numMwmIds);
 
-  if (type == RouterType::Pedestrian)
-  {
-    router = CreatePedestrianAStarBidirectionalRouter(indexGetterFn(), countryFileGetter, numMwmIds);
-    m_routingSession.SetRoutingSettings(routing::GetPedestrianRoutingSettings());
-  }
-  else if (type == RouterType::Bicycle)
-  {
-    router = CreateBicycleAStarBidirectionalRouter(indexGetterFn(), countryFileGetter, numMwmIds);
-    m_routingSession.SetRoutingSettings(routing::GetBicycleRoutingSettings());
-  }
-  else
-  {
-    auto & index = m_callbacks.m_featureIndexGetter();
-
-    auto localFileChecker = [this](std::string const & countryFile) -> bool {
-      MwmSet::MwmId const mwmId = m_callbacks.m_featureIndexGetter().GetMwmIdByCountryFile(
-          platform::CountryFile(countryFile));
-      if (!mwmId.IsAlive())
-        return false;
-
-      return version::MwmTraits(mwmId.GetInfo()->m_version).HasRoutingIndex();
-    };
-
-    auto const getMwmRectByName = [this](std::string const & countryId) -> m2::RectD {
-      return m_callbacks.m_countryInfoGetter().GetLimitRectForLeaf(countryId);
-    };
-
-    router = IndexRouter::CreateCarRouter(
-        countryFileGetter, getMwmRectByName, numMwmIds,
-        MakeNumMwmTree(*numMwmIds, m_callbacks.m_countryInfoGetter()), m_routingSession, index);
-    fetcher.reset(new OnlineAbsentCountriesFetcher(countryFileGetter, localFileChecker));
-    m_routingSession.SetRoutingSettings(routing::GetCarRoutingSettings());
-  }
+  auto & index = m_callbacks.m_featureIndexGetter();
+
+  auto localFileChecker = [this](std::string const & countryFile) -> bool {
+    MwmSet::MwmId const mwmId = m_callbacks.m_featureIndexGetter().GetMwmIdByCountryFile(
```

Comment:
Because the return value of the function is not a "feature index", but an "index".